src/components/Fields/TextField.js

```diff
@@ -265,19 +278,24 @@ class TextField extends Component {
         const icon = this.props.icon ?
             (
                 <span className="uir-text-field-left-icon">
-                    {this.props.icon}
+                    <Button
+                        className="icon-wrapper"
+                        icon={this.props.icon}
+                        onClick={this.handleClearIconClick}
+                        variant={ButtonVariant.CLEAR}
+                    />
                 </span>
             ) :
             null;
         const clearIcon = this.props.isClearable && this.state.value ?
             (
                 <span className="uir-text-field-right-icon">
                     <Button
+                        className="icon-wrapper"
```

Comment:
Should this class be prefixed with something?

Reply:
possibly... I initially thought `uir-text-field-icon-wrapper` was a bit long!! but perhaps that's the best way to go